### DISCUSSION TITLE
DataTable does not respect CSS alignment

### DISCUSSION THREAD

**Discussion by ofek** [Textualize/textual#1820] [Q&A] [answered]
```python
from textual.app import App
from textual.containers import Horizontal, Vertical
from textual.widgets import DataTable, Footer, Header, Label, Switch

class Listing(DataTable):
    def on_mount(self):
        super().on_mount()

        self.add_columns('foo', 'bar')
        self.add_row('a', 'b')

class MyApp(App):
    DEFAULT_CSS = """
    Vertical {
        width: 100%;
    }

    Horizontal {
        width: 100%;
        align: center middle;
    }
    """

    def compose(self):
        yield Header()
        yield Vertical(
            Horizontal(Switch()),
            Horizontal(Label(''.join(map(str, range(10))))),
            Horizontal(Listing()),
        )
        yield Footer()

if __name__ == '__main__':
    app = MyApp()
    app.run()
```

![a](https://user-images.githubusercontent.com/9677399/219513967-331fae26-89e5-4d39-bad2-80d22e5759d9.PNG)

**Comment by davep** [comment, marked as answer]
The `DataTable` is actually in the center/middle of the `Vertical` it's inside of and is following what you specified in CSS. If you're ever unsure about what's going on, a good chec is to place a border around the control in question. So, if you place a border around the `DataTable` in your example, you'll see:

![Screenshot 2023-02-17 at 09 05 45](https://user-images.githubusercontent.com/28237/219601096-bf2a7fa6-ecf6-497a-b18f-0db61c9ca21a.png)

In other words: it's the full width and height of the `Vertical` it's inside of. If you want it to visually appear more balanced you could do things like go with a border, specify a more appropriate width for it, perhaps set its background colour to raise it from the app's background, that sort of thing. For example, with this added to the CSS of your example:

```scss
    DataTable {
        background: $boost;
        border: round #444;
        width: 50%;
    }
```

you get this:

![Screenshot 2023-02-17 at 09 09 52](https://user-images.githubusercontent.com/28237/219602009-4151a221-d348-4e7f-9059-815f82f40f97.png)

Of course the other thing you can do is get a bit more creative with the column widths too; and at some point in the near future we hope to allow for some degree of relative column width sizing.

**Reply by willmcgugan** [reply]
This is a great tip. I usually do `border: solid red`

You could also set the width of the table to "auto" if you want the table to precisely line up in the center.
